### DISCUSSION TITLE
Serverless Function not recognising ts Import type

### DISCUSSION THREAD

**Discussion by martin-dimi** [vercel/community#2658] [Help] [answered]
### Summary

Hey, I've got a serverless function using langchain (openAI) dependency to make a call to chatGPT. I'm getting the following error:
```
node_modules/langchain/dist/base_language/count_tokens.d.ts:1:15 - error TS1005: ',' expected.
1 import { type TiktokenModel } from "js-tiktoken/lite";
```

It seems that the `type` keyword is not being recognised.  Seems to be running an old version of typescript, tho when I check `package-lock` the TS version is `4.9.3` which should support it.

I suspect my tsconfig maybe misconfigured but I'm struggling to pinpoint the issue...
tsconfig:
```
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "lib": ["ESNext", "DOM"],
    "moduleResolution": "Node",
    "strict": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "noEmit": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "skipLibCheck": true
  },
  "include": ["src"]
}
```

package.json:
```
{
  "name": "hey",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview"
  },
  "devDependencies": {
    "@types/uuid": "^9.0.1",
    "http-proxy-middleware": "^2.0.6",
    "typescript": "^4.9.3",
    "vite": "^4.2.0",
    "vite-plugin-vercel": "^0.1.7"
  },
  "dependencies": {
    "@cliqz/adblocker-playwright": "^1.26.5",
    "@medv/finder": "^3.0.0",
    "@vercel/node": "^2.10.3",
    "browser-editor": "file:../common/browserEditor",
    "chrome-aws-lambda": "^10.1.0",
    "cross-fetch": "^3.1.5",
    "langchain": "^0.0.84",
    "openai": "^3.2.1",
    "penpal": "^6.2.2",
    "playwright-core": "^1.32.3",
    "uuid": "^9.0.0"
  },
  "engines": {
    "node": "^20"
  }
}
```

### Steps to Reproduce

Using the config specified above 🙏

**Comment by amyegan** [comment, marked as answer]
Hi @martin-dimi! If you haven't already seen it, there is an [AI Agents with Langchain & OpenAI](https://vercel.com/templates/next.js/agent-gpt) template and open repo that might offer some clues about configuration and setup.

If you still need help with this, please share a minimal reproduction or link to a tutorial or template you used. Then we can run it and get a closer look at what is going wrong.

**Reply by martin-dimi** [reply]
No, actually, I haven't seen that one. Will check it out, thanks!
